### PR TITLE
GH-1400: Allow SERVICE QuerySendMode to be specified by a string

### LINKED ISSUE #1400: Allow the query send mode for SERVICE using a string in the context.
### Version

4.5.0

### Suggestion

SERVICE allows the send mode if the query request to be set but it is by giving a `QuerySendMode` enum.

Allow this to be a string as well.

In addition, provide "GET" and "POST" as additional names.

### Are you interested in contributing a solution yourself?

Yes

### PR DESCRIPTION
**Pull request by afs**
GitHub issue resolved #1400

Pull request Description:

Let the SERVICE send mode be set using a string.

----

 - [x] Tests are included.
 - [ ] Documentation change and updates are provided for the [Apache Jena website](https://github.com/apache/jena-site/)
 - [x] Commits have been squashed to remove intermediate development commit messages.
 - [x] Key commit messages start with the issue number (GH-xxxx or JENA-xxxx)

By submitting this pull request, I acknowledge that I am making a contribution to the Apache Software Foundation under the terms and conditions of the [Contributor's Agreement](https://www.apache.org/licenses/contributor-agreements.html).

----

See the [Apache Jena "Contributing" guide](https://github.com/apache/jena/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: jena-integration-tests/src/test/java/org/apache/jena/sparql/exec/http/TestService2.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    private static void runWithModifier(String key, HttpRequestModifier modifier, Runnable action) {
+        RegistryRequestModifier.get().add(SERVICE, modifier);
+        try {
+            action.run();
+        } finally {
+            RegistryRequestModifier.get().remove(SERVICE);
+        }
+    }
+}
```
Comment (kinow) [inline jena-integration-tests/src/test/java/org/apache/jena/sparql/exec/http/TestService2.java:142]: Missing newline :grimacing:

**Review by kinow — APPROVED** [review]
Feel free to ignore that comment :grimacing:

File: jena-integration-tests/src/test/java/org/apache/jena/sparql/exec/http/TestService2.java (hunk 1 above)
Comment (afs) [reply jena-integration-tests/src/test/java/org/apache/jena/sparql/exec/http/TestService2.java]: Done - and the file it was cloned from.

**Comment by afs** [comment]
It reuses an existing context key.

If you use `Service.java` then it will be honoured. If have a different execution code, you may want to implement the additional string value possibilities.

**Pull request merged**